src/clients/HubPoolClient.ts

```diff
@@ -409,9 +406,16 @@ export class HubPoolClient extends BaseAbstractClient {
       const tokenKey = `${deposit.originChainId}-${deposit.inputToken}`;
       if (this.l2TokenHasPoolRebalanceRoute(deposit.inputToken, deposit.originChainId, quoteBlockNumber)) {
         return (hubPoolTokens[tokenKey] ??= this.getL1TokenForDeposit({ ...deposit, quoteBlockNumber }));
-      } else return undefined;
+      }
+
+      return undefined;
     };
-    const getHubPoolTokens = (): EvmAddress[] => dedupArray(Object.values(hubPoolTokens).filter(isDefined));
+
+    // Filter hubPoolTokens for duplicates by reverting to their native string
+    // representation. This is required for deduplication to work reliably.
+    const getHubPoolTokens = (): EvmAddress[] => dedupArray(
+      Object.values(hubPoolTokens).filter(isDefined).map((token) => token.toNative())
+    ).map(EvmAddress.from);
```

Comment:
Driveby fix - it's not recommended to use `dedupArray()` on complex types, so translate the addresses to strings before deduplicating.